logger/common.go

```diff
@@ -173,100 +189,221 @@ func (l *Logger) sendLogs(
 		return err
 	}
 
-	scanner := bufio.NewScanner(f)
-	scanner.Split(bufio.ScanLines)
+	if err := l.read(ctx, f, source); err != nil {
+		err := errors.Wrapf(err, "failed to read logs from %s pipe", source)
+		debug.SendEventsToJournal(DaemonName, err.Error(), journal.PriErr, 1)
+		return err
+	}
+
+	debug.SendEventsToJournal(DaemonName,
+		fmt.Sprintf("Pipe %s is closed. Sleeping %s for cleanning up.", source, cleanupTime.String()),
+		journal.PriInfo,
+		0)
+	time.Sleep(*cleanupTime)
```

Comment:
Can you add a comment here for why this is needed?

Reply:
Sure, will add documentation here